File: spring-web-java/src/main/java/spring/web/java/domain/model/entity/MemberRole.java
package spring.web.java.domain.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.ConstraintMode;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class MemberRole extends Base {

	@Id
	@GeneratedValue
	@Column(name = "member_role_id")
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "member_id", foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT))
	private Member member;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "role_id", foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT))
	private Role role;

	public static MemberRole create(Role role) {
		MemberRole memberRole = new MemberRole();

		memberRole.role = role;

		return memberRole;
	}

	public void associateMember(Member member) {
		this.member = member;
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/entity/MemberRole.kt
package spring.web.kotlin.domain.model.entity

import jakarta.persistence.*

@Entity
class MemberRole protected constructor(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "role_id", foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    var role: Role,
) {
    @Id
    @GeneratedValue
    @Column(name = "member_role_id")
    var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    var member: Member? = null
        protected set

    companion object {
        fun create(role: Role) = MemberRole(role = role)
    }

    fun associateMember(member: Member) {
        this.member = member
    }
}